force-app/main/default/classes/RD2_ERecurringDonationsService.cls

```diff
@@ -154,4 +158,30 @@ public with sharing class RD2_ERecurringDonationsService {
         
         return frequency;
     }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2022
+    * @description process Recurring Donation record to build the status string.
+    * @return String returns the Status string.
+    */
+    private String getStatus(npe03__Recurring_Donation__c recurringDonation) {
```

Comment:
I think you could replace this with the following:
`RD2_RecurringDonation rd = new RD2_RecurringDonation(recurringDonation);`
`return rd.getState();`

Reply:
It's a processed State based on RD2_StatusMapper, Status__c field and the else case.